### ISSUE TITLE
Migrations happening in the wrong order

### ISSUE THREAD

**Issue by zeelot** [kohana-minion/tasks-migrations#4]
`./minion db:migrate --versions=FALSE` seems to be running the migrations by group instead of in the order they were created.

**Comment by bobeagan** [comment]
This is important.

**Comment by BRMatt** [comment]
I realise that, but I'm going to have to remove the part of the migrations model which allows you to migrate one location up and another down at the same time in order for this to work.

**Comment by BRMatt** [comment]
Here's an example of what I'm thinking of implementing, using example use cases

## Rename "Locations" to "Groups"

People always get confused as to what locations are, it makes more sense to call them groups and looks better if they're dot separated. (i.e. my.app instead of my/app)

## Replace `--versions` with `--migrate-up` `--migrate-down` `--migrate-to={version}`

The first two can be used on either a specific group or multiple groups, to avoid confusing situations the last one will only work on specific groups.

If migrating up or down then either `--group` or `--groups` can be used to specify an individual group / multiple groups (respectively) to migrate.

If migrating to a specific version then only `--group` can be used

## Examples

Migrating a _specific group_ up to the latest version

```
./minion db:migrate --group=app --migrate-up
```

Migrate _multiple groups_ up the latest version

```
./minion db:migrate --groups=app,asd --migrate-up
```

Migrating a _specific group_ all the way down (i.e. delete all tables created by the group)

```
./minion db:migrate --group=app --migrate-down
```

Migrating _multiple groups_ all the way down

```
./minion db:migrate --groups=app,asd --migrate-down
```

Migrate a _specific group_ to a _specific version_

```
./minion db:migrate --group=app   --migrate-to={version}
```

Where `{version} = ({timestamp}|+{up}|-{down})`

Thoughts?

**Comment by bobeagan** [comment]
This seems a lot cleaner to me.

**Comment by ThePixelDeveloper** [comment]
Agreed

**Comment by BRMatt** [comment]
Rewrite of the core migrations system is pretty much complete, closed by fd690e64940404fc241df75d995612763a6a5986
